nautobot_ssot/mixins.py

```diff
@@ -0,0 +1,175 @@
+                _vars[key] = str(db_obj.pk)
+            if key in list(self._many_to_many.keys()):
+                db_obj = get_fk(self._many_to_many, diffsync, key, val)
+                _vars[key] = str(db_obj.pk)
+
+        obj = db_model.objects.get(**_vars)
+
+        for key, value in instance_values.items():
+            if hasattr(self, "_skip") and key in self._skip:
+                continue
+            if key not in combined:
+                setattr(obj, key, value)
+            else:
+                if not value:  # need to find the value of the other object, if none exists, that won't work
+                    continue
+                if key in list(self._generic_relation.keys()):
+                    with_parent = {
+                        key: self._generic_relation[key]["parent"] for key in list(self._generic_relation.keys())
+                    }
+                    with_values = "__".join(
+                        [instance_values[key] for key in list(self._generic_relation[key]["identifiers"])]
+                    )
+                    db_obj = get_fk(with_parent, diffsync, key, with_values)
+                    setattr(obj, self._generic_relation[key]["attr"], db_obj)
+                if key in list(self._foreign_key.keys()):
+                    db_obj = get_fk(self._foreign_key, diffsync, key, value)
+                    setattr(obj, key, db_obj)
+                if key in list(self._many_to_many.keys()):
+                    if isinstance(value, list):
+                        for val in value:
+                            db_obj = get_fk(self._many_to_many, diffsync, key, val)
+                            getattr(obj, key).add(db_obj)
+                    else:
+                        db_obj = get_fk(self._many_to_many, diffsync, key, value)
+                        obj.set(db_obj)
+        obj.save()
+        return super().update(attrs)
+
+    def delete(self):
+        """Create Method to handle generically."""
+        db_model = self.Meta.model
+        obj = db_model.objects.get(pk=self.pk)
+        obj.delete()
+
+        super().delete()
+        return self
+
+
+class DiffSyncMixIn:
+    """Mixin to update add to allow for 'stuffing' data based on unique fields."""
+
+    _unique_data = {}
+
+    def add(self, obj, *args, **kwargs):
+        """Override add method to stuff data into dictionary based on the `_unique_fields`."""
+        super().add(obj, *args, **kwargs)
+        modelname = obj._modelname
+
+        for attr in getattr(obj, "_unique_fields", []):
+            if hasattr(obj, attr):
+                if not self._unique_data.get(modelname):
+                    self._unique_data[modelname] = {}
+                if not self._unique_data[modelname].get(attr):
+                    self._unique_data[modelname][attr] = {}
+                self._unique_data[modelname][attr][getattr(obj, attr)] = obj
+
+
+class AdapterMixIn:
+    def apply_diffsync_flags(self):
```

Comment:
isn't this something in diffsync? it seem to me like something for diffsync library